docroot/modules/custom/va_gov_backend/src/EventSubscriber/EntityEventSubscriber.php

```diff
@@ -49,6 +71,148 @@ public function entityPresave(EntityPresaveEvent $event): void {
+
+  /**
+   * Builds an array of descriptions from health services available on form.
+   *
+   * Adds the descriptions array built by this method to drupalSettings.
+   *
+   * @param \Drupal\core_event_dispatcher\Event\Form\FormIdAlterEvent $event
+   *   The event.
+   */
+  public function buildHealthServicesDescriptionArrayAddToSettings(FormIdAlterEvent $event): void {
+    $form = &$event->getForm();
+    $form_state = $event->getFormState();
+    $fields = $this->getProductTypeTermFields($form, $form_state);
+    $service_terms = $this->entityTypeManager
+      ->getListBuilder('taxonomy_term')
+      ->getStorage()
+      ->loadByProperties([
+        'vid' => 'health_care_service_taxonomy',
+      ]);
+    $descriptions = [];
+    foreach ($service_terms as $service_term) {
+      /** @var \Drupal\taxonomy\Entity\Term $service_term */
+      $descriptions[$service_term->id()] = [
+        'type' => $service_term->get($fields['type'])->getSetting('allowed_values')[$service_term->get($fields['type'])->getString()] ?? NULL,
+        'name' => $service_term->get($fields['name'])->getString(),
+        'conditions' => $service_term->get($fields['conditions'])->getString(),
+        'description' => trim(strip_tags($service_term->get($fields['description'])->value)),
+      ];
+    }
+    $form['#attached']['drupalSettings']['availableHealthServices'] = $descriptions;
+    $form['#attached']['library'][] = 'va_gov_backend/display_service_descriptions';
+  }
+
+  /**
+   * Builds an array of term fields predicated by product type.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array|null
+   *   The taxonomy fields.
+   */
+  public function getProductTypeTermFields(array &$form, FormStateInterface $form_state) {
+    $fields = [];
+    if ($form_state->getFormObject() instanceof EntityFormInterface) {
+      $bundle = $form_state->getFormObject()->getEntity()->bundle();
+      // Make the bundle available to displayServiceDescriptions.js.
+      $form['#attached']['drupalSettings']['currentNodeBundle'] = $bundle;
+      $fields = [
+        'type' => $bundle === 'vet_center' ? 'field_vet_center_type_of_care' : 'field_service_type_of_care',
+        'name' => $bundle === 'vet_center' ? 'field_vet_center_friendly_name' : 'field_also_known_as',
+        'conditions' => $bundle === 'vet_center' ? 'field_vet_center_com_conditions' : 'field_commonly_treated_condition',
+        'description' => $bundle === 'vet_center' ? 'field_vet_center_service_descrip' : 'description',
+      ];
+      return $fields;
+    }
+    return NULL;
+  }
+
+  /**
+   * Appends health service entity description to title on entity view page.
+   *
+   * @param \Drupal\core_event_dispatcher\Event\Entity\EntityViewAlterEvent $event
+   *   The entity view alter service.
+   */
+  public function appendHealthServiceTermDescription(EntityViewAlterEvent $event):void {
```

Comment:
medium: Since this function only seems to operate on Vet Centers, it would be nice to make that clear in the name.
 ```suggestion
  public function appendHealthServiceTermDescriptionToVetCenter(EntityViewAlterEvent $event):void {
```

Reply:
Yeah, fair enough. I was kinda thinking this would be generic enough for future use for other content types, but I take your point, @swirtSJW :+1: